src/plugins/vis_type_xy/public/utils/get_color_picker.tsx

```diff
@@ -0,0 +1,96 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+
+import { LegendColorPicker, Position, XYChartSeriesIdentifier, SeriesName } from '@elastic/charts';
+import { PopoverAnchorPosition, EuiWrappingPopover, EuiOutsideClickDetector } from '@elastic/eui';
+import type { PersistedState } from '../../../visualizations/public';
+import { ColorPicker } from '../../../charts/public';
+
+function getAnchorPosition(legendPosition: Position): PopoverAnchorPosition {
+  switch (legendPosition) {
+    case Position.Bottom:
+      return 'upCenter';
+    case Position.Top:
+      return 'downCenter';
+    case Position.Left:
+      return 'rightCenter';
+    default:
+      return 'leftCenter';
+  }
+}
+
+const KEY_CODE_ENTER = 13;
+
+export const getColorPicker = (
+  legendPosition: Position,
+  setColor: (newColor: string | null, seriesKey: string | number) => void,
+  getSeriesName: (series: XYChartSeriesIdentifier) => SeriesName,
+  paletteName: string,
+  uiState: PersistedState
+): LegendColorPicker => ({
+  anchor,
+  color,
+  onClose,
+  onChange,
+  seriesIdentifiers: [seriesIdentifier],
+}) => {
+  const seriesName = getSeriesName(seriesIdentifier as XYChartSeriesIdentifier);
+  const overwriteColors: Record<string, string> = uiState?.get('vis.colors', {});
+  const colorIsOverwritten = Object.keys(overwriteColors).includes(seriesName as string);
+  let keyDownEventOn = false;
+
+  const handleChange = (newColor: string | null) => {
+    if (!seriesName) {
+      return;
+    }
+    if (newColor) {
+      onChange(newColor);
+    }
+    setColor(newColor, seriesName);
+    // close the popover if no color is applied or the user has clicked a color
+    if (!newColor || !keyDownEventOn) {
+      onClose();
+    }
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
+    if (e.keyCode === KEY_CODE_ENTER) {
+      onClose?.();
+    }
+    keyDownEventOn = true;
+  };
+
+  const handleOutsideClick = useCallback(() => {
+    onClose?.();
+  }, [onClose]);
+
+  return (
+    <EuiOutsideClickDetector onOutsideClick={handleOutsideClick}>
+      <EuiWrappingPopover
+        isOpen
+        ownFocus
+        display="block"
+        button={anchor}
+        anchorPosition={getAnchorPosition(legendPosition)}
+        closePopover={onClose}
+        panelPaddingSize="s"
+      >
+        <ColorPicker
+          color={paletteName === 'kibana_palette' ? color : color.toLowerCase()}
+          onChange={handleChange}
+          label={seriesName}
+          useLegacyColors={paletteName === 'kibana_palette'}
+          colorIsOverwritten={colorIsOverwritten}
+          onKeyDown={onKeyDown}
+        />
+      </EuiWrappingPopover>
```

Comment:
Not sure how this was affected, but there seems to be no focus on the color dots when you using keyboard actions to open the color picker.

Below I focus a legend item and then tab to the color, hit enter and then tabbing when the color picker is open does nothing. I'm pretty sure this was working before.

https://user-images.githubusercontent.com/19007109/110849085-707d2900-8274-11eb-912e-a523626b28b8.mp4

@myasonik might have already mentioned this above. Not sure how to fix it.

Reply:
The dots are now a radio group so arrow keys move you back and forward rather than tab.

If we think this is tricky for users at large, we can think about alternatives but I think this is better than where the component was at previously.

Reply:
Oh I see, that's fine. Thanks for the explanation. Yeah would be nice to have an initial state to know it focused on the popover.

Reply:
What do you mean to focus on the first dot when it opens? I don't think we want this. We want to focus on the active color. Like here:
![image](https://user-images.githubusercontent.com/17003240/110965434-da70fd80-835c-11eb-8d39-99e18c361e23.png)
 
On your screenshot below @myasonik there is the legacy palette. The current series color doesn't exist on the popover so for this reason you don't see an active dot.

I think it is wrong to focus on the first dot, this means that every time I open the popover, the first color is selected and the chart will automatically change color. We def don't want this.

Reply:
Currently there is a `:focus` state and I think an `:active` or selected state. I think we should add back the focus state and auto focus the first color on focus of the color picker. See the recording below.


https://user-images.githubusercontent.com/19007109/111168084-ba655800-856f-11eb-9032-c93a77d2bba0.mp4

Reply:
This was lost due to our change on custom input radio btns. I could add it with css but I am not sure about it. If you see Wylie's comment above, previously we had auto focus on the first element, which was quite confusing. I will sync with @myasonik offline to decide what we want to do here. Maybe the previous implementation with tab navigation was better, if we take under consideration that this popover is going to change in the future.